loopy/kernel/__init__.py

```diff
@@ -314,6 +317,19 @@ def __init__(self, domains, instructions, args=None,
             from loopy.kernel.tools import SetOperationCacheManager
             cache_manager = SetOperationCacheManager()
 
+        inames_from_domains = set().union(*
+                (set(dom.get_var_names(dim_type.set)) for dom in domains))
+
+        if inames is None:
+            inames = frozenset(Iname(name) for name in inames_from_domains)
+        else:
+            assert set(iname.name for iname in inames) == inames_from_domains
+
+        if iname_to_tags is not None:
+            inames = frozenset(
+                    iname.copy(tags=iname_to_tags.get(iname.name, frozenset()))
+                    for iname in inames)
```

Comment:
Disallow passing both `inames` and `iname_to_tags`.

Reply:
Done 8b3131e